Prepare CHANGELOG for 1.0.2 release (#2482)
spiffe/spire: CHANGELOG.md

# Changelog

## [1.0.2] - 2021-09-02

### Added
- Experimental support for custom authorization policies based on Open Policy Agent (OPA) (#2416)
- SPIRE Server can now be configured to emit audit logs (#2297, #2391, #2394, #2396, #2442, #2458)

> - SDS v3 API in agent now supports the SPIFFE Certificate Validator (#2435, #2460)

Reviewer: A little bit more about what we get, and that it is related to envoy, could be good here.

```suggestion
- Envoy SDS v3 API in agent now supports the SPIFFE Certificate Validator for federated SPIFFE authentication (#2435, #2460)
```

> - Enhancements in the SPIRE OIDC Discovery Provider (#2404, #2453)

Reviewer: There were a couple changes with these PRs but I think these two were the main ones.

```suggestion
- SPIRE OIDC Discovery Provider can now listen on a unix domain socket (#2404)
- SPIRE OIDC Discovery Provider now intelligently handles host headers (#2404, #2453)
```

Reviewer: @amartinezfayo pointed out that the provider already supported UDS

```suggestion
- SPIRE OIDC Discovery Provider now intelligently handles host headers (#2404, #2453)
```
